controllers/datadogagent/override/dependencies.go

```diff
@@ -12,20 +12,48 @@ import (
 
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/configmap"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
 )
 
 // Dependencies is used to override any resource/dependency settings with a v2alpha1.DatadogAgentComponentOverride.
 func Dependencies(logger logr.Logger, manager feature.ResourceManagers, overrides map[v2alpha1.ComponentName]*v2alpha1.DatadogAgentComponentOverride, namespace string) (errs []error) {
 	for component, override := range overrides {
-		err := overrideComponentDependencies(logger, manager, override, component, namespace)
+		err := overrideRBAC(logger, manager, override, component, namespace)
 		if err != nil {
 			errs = append(errs, err)
 		}
+
+		// Handle custom check configurations, only if ConfigMap == nil
+		if override.ExtraConfd != nil && override.ExtraConfd.ConfigMap == nil && len(override.ExtraConfd.ConfigDataMap) > 0 {
+			cm, err := configmap.BuildConfigMapMulti(namespace, override.ExtraConfd.ConfigDataMap, v2alpha1.ExtraConfdConfigMapName, true)
+			if err != nil {
+				errs = append(errs, err)
+			}
+			if cm != nil {
+				if err := manager.Store().AddOrUpdate(kubernetes.ConfigMapKind, cm); err != nil {
+					errs = append(errs, err)
+				}
+			}
+		}
```

Comment:
nits: this block can be moved to a function, because we have the same exact block for ExtraChecksd